[18.0][MIG] web_responsive: Migration to 18.0

chaule97 opened the pull request:
- The features this module brings compared to native:
   - New navigation with the fullscreen app menu
   - Quick menu search inside the app menu
   - Sticky header & footer in list view
   - Sticky statusbar in form view
   - Bigger checkboxes in list view
   - Followers and send button is displayed on mobile. Avatar is hidden.
   - Big inputs on form in edit mode
   - Keyboard shortcuts for easier navigation, **using \`Alt + Shift + \[NUM\]\`**
   - Autofocus on search menu box when opening the app menu
   - When the chatter is on the side part, the document viewer fills that part for side-by-side reading instead of full screen
   - When the user chooses to send a public message the color of the composer is different from the one when the message is an internal log.

- any changes in v18:
    -`Control panel buttons use icons to save space.` function is removed 
![image](https://raw.githubusercontent.com/OCA/web/17.0/web_responsive/static/img/form_buttons.gif)

because odoo 18 changed template
![image](https://github.com/user-attachments/assets/c144687b-e3b3-4ce5-b765-a96c339956a2)

lasley committed:
[FIX] web_responsive: Other minor fixes
* Add Tecnativa as author
* Remove untestable JS method
* Rename to web_responsive

lasley committed:
[MIG] web_responsive: Migrate to v10
* Bump version
* Change all openerp to odoo
* Fix qunit suite injection
* Remove jQuery require
* Change to new selectors:
  * `oe_leftbar` to `o_sub_menu`
  * `oe_logo` to `o_sub_menu_logo`
  * `oe_footer` to `o_sub_menu_footer`
  * `oe_secondary_menus_container` to `o_sub_menu_content`
* Add style to hide oe_footer instead of remove to not break `support_branding`
* Add note in ReadMe explaining override of `support_branding`
* Set top margin of app drawer title to 0 to fit v10 proportions
* Fix notification badge positioning
* Add o_web_client class to body to fix overlaps
* Scroll control panel with page
* Change navbar z-index to not be overlapped by buttons
* Raise z-index on header to raise over buttons
* Handle layered notifications via z-index
* Remove `#` from navigation links HREF to prevent history littering

lasley committed:
[FIX] web_responsive: Change z-indexes
* Rebalance z-indexes to fix overlay issue while still providing notification support

oca-transbot committed:
Update translation files

danieltorres7 committed:
Form view improvements to fit smaller screens

On screens with less than 768px content is limited to 2 columns.

On mobile field labels have their own line.

Fix elements width for XS screens

danieltorres7 committed:
Form image editing controls always present

Make image editing controls always available, instead of depending
on resolution or hover.

danieltorres7 committed:
Web responsive menu improvements (#513)

* Bold titles for apps in the app-drawer
* Remove z-index from Control Panel buttons, so it doesn't overlap the menu.
* Better responsive menus
  * Move systray icons/menus out of the menu and up to the top bar.
  * Increase avatar size and visibility.

* Other minor menu behavior fixes.
* Changes to fit odoo variables
* And a little margin adjust on navbar-right
* Top menu always thick
* Removed overflow:hidden!important from main.less
* Local patch: prevent body overflow auto
* Stop using `style=` on `<body>`
* Fix jquery drawer in anticipation of:
   https://github.com/blivesta/drawer/pull/36

yajo committed:
Fix problem with overflowing chatter widget.

yajo committed:
[FIX][web_responsive] Form padding, titles and buttons OK in XS

Small makeup to make it look good in XS screens.

sergio-teruel committed:
[10.0][IMP] web_responsive: Load default kanban view for small screen

Commit:
[10.0][IMP][web_responsive] (#569)

* [IMP] drawer-toggle

* [IMP] Hitbox of close drawer-toggle

* [IMP] Removed outline on menu item

* [IMP] Logo positionings

* [FIX] Try me on runbot button

* [REM] Reversed outline change

zeroheure committed:
Update README to fix wikipedia link

lasley committed:
[FIX] web_responsive: Change edit accesskey
* Change accesskey for `edit` in form view back to `e` to fix #587

yajo committed:
[FIX][web_responsive] Make it compatible with l10n_es_toponyms

Before this patch, there was an incompatibility between this addon and l10n_es_toponyms, caused by the lack of wrapping of the special address field that was being added.

Now layout works as expected in that case. In any other case, it seems sensible to wrap inputs so no more weird overflows happen and less chances of needing horizontal scrolling happen.

pedrobaeza committed:
[FIX] web_responsive: Execution permission

Commit:
[10.0][IMP][web_responsive] Make search bar sticky  (#633)

* Initial commit

* Initial commit

yajo committed:
[FIX][web_responsive] Long-named multicompanies nice in XS (#688)

lasley committed:
[FIX] web_responsive: Chrome Slow Scrolling
* Update to iScoll bleeding edge to fix #741

angelmoya committed:
UPDT version

angelmoya committed:
FIX travis (#747)

* FIX hasclass

* FIX views

* [FIX] web_responsive: Syntax error in xpath

pedrobaeza committed:
[FIX] web_responsive: Correct runbot link

hgminh95 committed:
[MIG] web_responsive test to 11.0

lasley committed:
[FIX] web_responsive: Menu Height on Init
* Add a call to the window resize method during init to fix OCA#773

fanha99 committed:
[FIX] web_responsive: Hide header when print directly from browser issues 808 (#809)

Closes #808

lasley committed:
[IMP] web_responsive: Add search feature (#838)

* Add feature to search menus in the app drawer
* Switch model to RPC

Tardo committed:
[IMP] web_responsive: Possibility of putting the chatter in the right  (#951)

OCA-git-bot committed:
[UPD] README.rst

Tardo committed:
[FIX] Chatter Position

Tardo committed:
[FIX] Read/Write permissions

Tardo committed:
[CLEAN] Comments

Tardo committed:
[IMP] Code

Tardo committed:
[IMP] Adding res.users chatter_position read/write test

yajo committed:
Fix linter

Tardo committed:
[IMP] Sticky Statusbar

Tardo committed:
[IMP] Add sticky header & footer in list view

Tardo committed:
[FIX][11.0][web_responsive] Views without sheet

yajo committed:
[FIX] web_responsive: Complete search feature forward-porting

Tardo committed:
[IMP][11.0][web_responsive] Suggested changes (#978)

* [IMP] Chatter Position
* [IMP] Clean unused stuff
* [IMP] FieldStatus hidden-xs
* [IMP] Improvements suggested in #540
* [WIP] Suggested changes
* [WIP] Improve FieldStatus-Responsive
* [IMP] Statusbar Buttons & FieldStatus
* [IMP] Statusbar Buttons Style
* [IMP] Button 'Task' style

yajo committed:
[FIX] web_responsive: Disable access to hidden menus

When doing menu search, some menus that the user didn't have permission to use were presented to him.

Fix #850.

pedrobaeza committed:
[IMP] web_responsive: Add known issue note about sticky

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

yajo committed:
[MIG] web_responsive: Migrate to v12 and refactor

This migration includes a full refactoring to make this module more
maintainable. Some things that have changed:

- Removed external libraries.
- Change Less for Scss.
- Reduce ES and XML to the minimal required needs.
- Implement as much features as possible with just Scss.
- Remove copyright from `__init__.py` files.
- Trigger the new hotkeys system from Odoo v12 with `Shift+Alt` instead
  of just `Alt`, and restore some good old hotkeys (`E` for "Edit",
  `D` for "Discard", and `A` for "Apps menu").
  See https://github.com/odoo/odoo/issues/30068 on the matter.
- Control panel breadcrumbs are collapsed into a single backwards icon.
- Add FA icons to most common buttons in control panel.
- Hide text in XS for those buttons, to have a slicker phone experience.
- Lots of gifs in the README!

oca-travis committed:
Update translation files

yajo committed:
[FIX] web_responsive: Skip re-search when not writing

When the user uses TAB or SHIFT+TAB to navigate in search results, he can trigger a SHIFT-only keydown event without noticing, which would reset the selected result position to the first or last.

It is disturbing, so, to avoid that problem, the re-search is only triggered if the currently pressed key has length=1 or is Backspace, which will skip most keys that are not actually writing a character into the search input, like i.e. "Shift", "Alt", "F4", etc.

In addition to that, to detect if the search results is empty, the `:empty` selector is not trustable because it considers not empty nodes with whitespace-only content. That has been patched too.

yajo committed:
[FIX] web_responsive: Make app icons and search results middle-button-clickable

In the case of app icons, those weren't getting the `href` attribute filled.

In the case of search results, those weren't even an `<a>` element.

Fixed in both cases to fix https://github.com/OCA/web/issues/1170.

Tardo committed:
[IMP] Sticky chatter topbar

Tardo committed:
[FIX+IMP] web_responsive: Chatter Topbar + fix #1189 + added gif

yajo committed:
Update web_responsive/static/src/css/web_responsive.scss

Co-Authored-By: Tardo <dev@redneboa.es>

yajo committed:
[FIX] web_responsive: Allow searching apps in mobile Chrome

This patch fixes #1251.

The keydown event is faked in mobile chrome because it's not something reliable due to today's virtual keyboards prediction features. Typing a <kbd>C</kbd> character doesn't mean it will be there when user uses autocorrection/autoguessing.

Instead of that, we use now the `keydown` event exclusively for movement features, and `input` event exclusively for content changes in the search input element.

Apart from fixing search in mobile Chrome, it also makes code more readable.

The search input also disables autocompletion since it would be useless and annoying here.

Tardo committed:
[IMP] web_responsive: chatter width

Tardo committed:
[IMP] web_responsive: Sticky in tree view and form view

Tardo committed:
[IMP] web_responsive: Support long title in form view

Tardo committed:
[IMP] web_responsive: AppMenu wait for load view

Tardo committed:
[IMP] web_responsive: Suggested changes

Tardo committed:
[IMP] web_responsive: Menu wait for action

Tardo committed:
[IMP] web_responsive: Readme

Tardo committed:
[FIX] web_responsive: AppDrawer in Chromium

Tardo committed:
[IMP] web_responsive: Refactor code

Tardo committed:
[IMP] web_responsive: Readme

Tardo committed:
[FIX] web_responsive: Chatter position

Tardo committed:
imp overflow

Tardo committed:
[IMP] web_responsive: Chatter on small screen

Tardo committed:
[FIX] web_responsive: Close AppDrawer

keshrath committed:
[FIX] web_responsive: AppDrawer Discard

keshrath committed:
[FIX] web_responsive: adapt AppDrawer Discard

keshrath committed:
[FIX] web_responsive: Close Menu on Discard

OCA-git-bot committed:
[UPD] README.rst

keshrath committed:
Update web_responsive.scss

Tardo committed:
[IMP] web_responsive: Performance when hide menus

Tardo committed:
[IMP] web_responsive: Don't show menu if waiting for an action to end

Tardo committed:
[IMP] web_responsive: Small refactor 'isWaiting'

Tardo committed:
[FIX] web_responsive: changed button box more initial position

Tardo committed:
[MIG] web_responsive: Migration to 13.0

Tardo committed:
[IMP] web_responsive: Made chatter input & attachments scrollable

Tardo committed:
[IMP] web_responsive: Support long titles

Tardo committed:
black, isort

pedrobaeza committed:
[IMP] web_responsive: Add development status + contributors

OCA-git-bot committed:
[UPD] README.rst

emagdalenaC2i committed:
[FIX] web_dialog_size js warnings

Tardo committed:
[IMP] web_responsive: Adapt test to 13.0

Tardo committed:
[IMP] mail_tracking: Changed form statusbar z-index

Tardo committed:
[IMP] web-responsive: Document Viewer

Commit:
[FIX] web_responsive: make followers dropdown usable

Fix a visual glitch that impeded using the followers dropdown with a sided chatter.

Commit:
[IMP] pre-commit run after update

Includes some manual fixes to silent ESLint warnings.

Tardo committed:
[FIX] web_responsive: Change AppDrawer close method

Previous of this commit, AppDrawer closes when the action have an id
that is defined in a menu item and no rpc calls are running.
Now AppDrawer closes in a more generic way, is enough that a
controller is attached to consider that the AppDrawer needs to be
closed.

sbidoul committed:
pre-commit update: prettier xml indent

kos94ok-3D committed:
[FIX] web_responsive: fix hotkey combinations in a shortcut template

Tardo committed:
[IMP] web_responsive: Restore discuss mobile mode

androettop committed:
New style in apps menu

scss file to change ui variables easily

Deleted unnecesary padding

remove unnecesary styles

unnecesary styles

align menu items to left again

search results styles and fixes

prettier configuration fix

newline at end of file (variables.scss)

text color calculation for navbar

fix in img url

requested fixes

Transitions and focus fixes

fix in js settimeout

margin added in search-container

Fix in transition

Co-authored-by: Jairo Llopis <Yajo@users.noreply.github.com>

Remove duplicated css

ivantodorovich committed:
[IMP] Improve responsiveness in large screens: don't let the icons separate too much from each other

[FIX] Display of long search result items in mobile

[IMP] AppMenu searchbar style

[IMP] Add vertical padding to icons, to increase separation

[IMP] Add focus style on app icons

[IMP] Prevent user selection on menu

[IMP] Prevent dragging the menues

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[IMP] web_responsive: Adjust roadmap

* Removed one of the wishlist after #1654 and #1657
* Added the bug depicted in #1626

Closes #1626

OCA-git-bot committed:
[UPD] README.rst

Tardo committed:
[FIX] web_responsive: Hide search panel on small screens

Tardo committed:
[IMP] web_responsive: Cut long filters names in the filters menu

Tardo committed:
[IMP] web_responsive: Apply pre-commit changes

Tardo committed:
[FIX] web_responsive: Menu section overflow

Tardo committed:
[IMP] web_responsive: Bigger checkboxes in list view

ypapouin committed:
[FIX] web_responsive: Do not set overflow property on inline fields
Co-authored-by: Alexandre Díaz <alexandre.diaz@tecnativa.com>
Ensure that the title is displayed while the user holds the mouse click (while selecting content)

Commit:
[MIG] web_responsive: Migration to 14.0

Tardo committed:
[IMP] web_responsive: Scrollable dropdowns

oca-travis committed:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[UPD] README.rst

SplashS committed:
[FIX] error on grouped mobile kanban with zero groups

SplashS committed:
Added translation using Weblate (Russian)

SplashS committed:
Translated using Weblate (Russian)

Currently translated at 87.5% (21 of 24 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/ru/

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 50.0% (12 of 24 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fr_FR/

SplashS committed:
[FIX] word wrap for appdrawer menu items

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 77.7% (14 of 18 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/nl/

remi-filament committed:
Translated using Weblate (French)

Currently translated at 77.7% (14 of 18 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fr/

saeed-raeisi committed:
Added translation using Weblate (Persian)

saeed-raeisi committed:
Translated using Weblate (Persian)

Currently translated at 100.0% (24 of 24 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fa/

saeed-raeisi committed:
Added translation using Weblate (Persian)

remi-filament committed:
Translated using Weblate (French)

Currently translated at 88.8% (16 of 18 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fr/

SplashS committed:
[MIG] web_responsive: Migration to 15.0

SplashS committed:
[IMP] show AppsMenu by default after login
also redirect to backend from website AppsMenu

SplashS committed:
[IMP] recaptured actual description screenshots

Tardo committed:
[IMP] web_responsive: Adapt label width

SplashS committed:
[IMP] big inputs on forms in edit mode

Tardo committed:
[FIX] web_responsive: Label width on modal

SplashS committed:
[FIX] make base odoo JS tests working with web_responsive module

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[IMP] web_responsive: Indicate incompatibility

Closes #2149

tate11 committed:
Added translation using Weblate (Turkish)

SplashS committed:
[FIX] kanban mode for popup in Many2One fields on mobile

SplashS committed:
[FIX] redundant padding on kanban card

OCA-git-bot committed:
web_responsive 15.0.1.1.0

hieulucky111 committed:
[FIX]-web_responsive module: Go to wrong action menu when user is set with Home Action menu

OCA-git-bot committed:
web_responsive 15.0.1.1.1

jacekmichalski committed:
Added translation using Weblate (Polish)

jacekmichalski committed:
Translated using Weblate (Polish)

Currently translated at 91.6% (22 of 24 strings)

Translation: web-15.0/web-15.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-15-0/web-15-0-web_responsive/pl/

SplashS committed:
[FIX] web_responsive: make keyboard navigation in Apps menu Search results works again.
Due to changes in javascript framework keyboard navigation was broken.
This commit makes it work again.

SplashS committed:
[FIX] web_responsive: attribute 'records_draggable' was no effect on desktop.
Kanban View has an attribute 'records_draggable', means 'whether it should be possible to drag records when kanban is grouped'.
It's support was broken.
Co-authored-by: MackZhong <mackz@sina.com>

OCA-git-bot committed:
web_responsive 15.0.1.1.2

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
web_responsive 15.0.1.1.3

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-15.0/web-15.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-15-0/web-15-0-web_responsive/

vincent-hatakeyama committed:
Translated using Weblate (French)

Currently translated at 100.0% (24 of 24 strings)

Translation: web-15.0/web-15.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-15-0/web-15-0-web_responsive/fr/

SplashS committed:
[FIX] web_responsive: fix dropdown caret visibility on website

before this commit dropdown on website doesn't have caret visible. F.x. to show option "Edit on backend"

after this commit caret will be visible

OCA-git-bot committed:
web_responsive 15.0.1.1.4

NICO-SOLUTIONS committed:
Translated using Weblate (German)

Currently translated at 54.1% (13 of 24 strings)

Translation: web-15.0/web-15.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-15-0/web-15-0-web_responsive/de/

vincent-hatakeyama committed:
[FIX] web_responsive: only change width of main control panel

The Search More panel is incorrectly displayed due to the reduced screen
so only change width on the main screen.

Fixes #2395

OCA-git-bot committed:
web_responsive 15.0.1.1.5

bvanjeelharia committed:
[15.0][FIX] Fix problem of blank page when using web_responsive

Changes for default page open

OCA-git-bot committed:
web_responsive 15.0.1.1.6

bvanjeelharia committed:
[MIG] web_responsive: Migration to 16.0

Fix for optional dropdown checkbox in list view

Made Changes for compatibility with web_chatter_position module

Fix for attachment delete dialog

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 87.5% (21 of 24 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/tr/

bvanjeelharia committed:
[FIX] web_responsive: Align checkboxes in Favourites save current search view

bvanjeelharia committed:
[FIX] web_responsive: Align menu for dates in filters

OCA-git-bot committed:
web_responsive 16.0.1.0.1

bvanjeelharia committed:
[FIX] web_responsive: Show View Type Always

OCA-git-bot committed:
web_responsive 16.0.1.0.2

OCA-git-bot committed:
web_responsive 16.0.1.0.3

remi-filament committed:
Translated using Weblate (French)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/fr/

Hung412 committed:
[FIX] web_responsive: fix bug test ESLint

OCA-git-bot committed:
web_responsive 16.0.1.0.4

bvanjeelharia committed:
[FIX] web_responsive: Translation Button Position

OCA-git-bot committed:
web_responsive 16.0.1.0.5

hugosantosred committed:
[FIX] Split button templates to show translated labels

Fixes #2439

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
web_responsive 16.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/hr/

CarlosRoca13 committed:
[FIX] web_responsive: Wrap text when display_name is too long

CarlosRoca13 committed:
[IMP] web_responsive: Remove z and x hotkeys

OCA-git-bot committed:
web_responsive 16.0.1.1.1

OCA-git-bot committed:
web_responsive 16.0.1.1.2

NICO-SOLUTIONS committed:
Translated using Weblate (German)

Currently translated at 94.4% (17 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/de/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/es/

OCA-git-bot committed:
[UPD] README.rst

chienandalu committed:
[IMP] web_responsive: public message chatter color

We give a color clue to the users so they can better tell if they're
sending an internal note or a public message.

Dislaimer: we use the `has()` pseudo class mainly because is simpler to
implement and the support is quite extended right now. Even Firefox will
catch up soon (now it has support behing a config flag).

So users with compatible browser will enjoy it, while others will just stay
as they were.

TT44953

kikopeiro committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/es/

OCA-git-bot committed:
oca-github-bot post-merge updates

Commit:
[16.0][IMP] web_responsive: To follow the app menu icon in Odoo 16

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[16.0][FIX] web_responsive: Fixed full-screen apps menu

OCA-git-bot committed:
[BOT] post-merge updates

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 88.8% (16 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/pt_BR/

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (18 of 18 strings)

Translation: web-16.0/web-16.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_responsive/pt_BR/

hafizhlf committed:
[FIX] web_responsive: Update z-index in apps menu.scss to fix overlapping issue:

- Increase z-index from 1000 to 10000 in apps_menu.scss.

OCA-git-bot committed:
[BOT] post-merge updates

hafizhlf committed:
Revert "[FIX] web_responsive: Update z-index in apps menu.scss to fix overlapping issue:"

This reverts commit 5de47a859287bbd71cd0532a739a0f943d6384ad.

hafizhlf committed:
[16.0][web_responsive] refactor: Update web_responsive.scss

- set z-index for .o_searchview_autocomplete element.

OCA-git-bot committed:
[BOT] post-merge updates

TesaLazarenko committed:
[MIG] web_responsive: Migration to 17.0

yajo committed:
[BUILD] web_responsive: resign as maintainer

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/

mymage committed:
Added translation using Weblate (Italian)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (29 of 29 strings)

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (29 of 29 strings)

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/it/

TesaLazarenko committed:
[17.0][FIX] web_responsive: disable default file preview

OCA-git-bot committed:
[BOT] post-merge updates

Patelangel1995 committed:
[IMP] web_responsive: Redirect to home after login
Redirect to the home page after login will occur only if the user has enabled the 'Redirect to Home' configuration in their user profile settings

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (31 of 31 strings)

Translation: web-17.0/web-17.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-17-0/web-17-0-web_responsive/it/

CarlosRoca13 committed:
[FIX] web_responsive: Can't access web using superuser

Before this change when trying to access web using superuser the web
interface is not charged.

By doing this change is all working normally.

Steps to reproduce the problem:

1. Log in
2. Go to web/become

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[FIX] web_responsive: Fix test related to ir.actions.act_window

Related to https://github.com/odoo/odoo/commit/cac20c5f2f0da7be6a75bd1281d528df7f34bfe6

Now it is necessary to have an ir.actions.act_window record

OCA-git-bot committed:
[BOT] post-merge updates

kafai-lam committed:
[FIX] web_responsive: Add support to serve menu icon in image url

Add support to serve menu icon in image URL format rather than base64.

It can take advantanges from browser caching and Nginx proxy.

Signed-off-by: Fai <lamkafai1997@gmail.com>

OCA-git-bot committed:
[BOT] post-merge updates

chaule97 committed:
[IMP] web_responsive: pre-commit auto fixes

chaule97 committed:
[MIG] web_responsive: Migration to 18.0

HviorForgeFlow reviewed (changes requested):
Functional review.

Click on an entry that has been found from a search in the menu link is broken. It uses old way action_id=123 but keeps the /odoo/action-321 old (for instance http://localhost:8069/odoo/action-123?action_id=321&menu_id=456). I guess that the href should be build with the new reference.

chaule97 commented:
> Functional review.
> 
> Click on an entry that has been found from a search in the menu link is broken. It uses old way action_id=123 but keeps the /odoo/action-321 old (for instance http://localhost:8069/odoo/action-123?action_id=321&menu_id=456). I guess that the href should be build with the new reference.

Thanks, I have fixed it

kobros-tech commented on web_responsive/tests/test_ir_http.py:
```diff
@@ -0,0 +1,34 @@
+from odoo.tests import HttpCase
+
+
+class TestIrHttp(HttpCase):
```
Is there a reason to remove this tag:
`@tagged("-at_install", "post_install")`

chaule97 replied:
because of this error:
`File "/opt/odoo/odoo/sql_db.py", line 572, in execute return self._cursor.execute(*args, **kwargs) File "/opt/odoo/odoo/sql_db.py", line 354, in execute res = self._obj.execute(query, params) psycopg2.errors.UndefinedColumn: column res_users.user_year_born does not exist LINE 1: ...apps_menu_theme", "res_users"."is_redirect_home", "res_users...` in module `web_pivot_computed_measure`

So, I think testcase in `web_responsive` should be runned first

kobros-tech reviewed (changes requested):
there is a test file and you removed it too, here is the content:

`* Copyright 2021 ITerra - Sergey Shebanin
 * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
odoo.define("web_responsive.test_patch", function (require) {
    "use strict";

    const utils = require("web_tour.TourStepUtils");

    /* Make base odoo JS tests working */
    utils.include({
        showAppsMenuItem() {
            return {
                edition: "community",
                trigger: ".o_navbar_apps_menu",
                auto: true,
                position: "bottom",
            };
        },
    });
});`

web_responsive/static/tests/test_patch.js

could you explain, please?

SplashS commented:
> web_responsive/static/tests/test_patch.js


This file was necessary for testing engine in general.
Odoo testing framework in part of browser integration tests (post_install) make clicks.
And it also wants to click on the main menu.
But web_responsive module changes the main menu and the way to click it programmatically.
This file allowed post_install tests to work.

pedrobaeza commented:
There's already #3037 doing the migration.

pilarvargas-tecnativa committed:
[IMP] web_responsive: Remove unnecessary calendar optimisation

Removed the optimisation of hiding the week number at a certain size
for consistency with the responsive design of Odoo enterprise. It is a
non-functional optimisation as it hides useful information that hardly
takes up any space. It also causes incompatibilities in the design with
extensions that do not take into account these specific changes.

TT54811

chaule97 commented:
> > web_responsive/static/tests/test_patch.js
> 
> This file was necessary for testing engine in general. Odoo testing framework in part of browser integration tests (post_install) make clicks. And it also wants to click on the main menu. But web_responsive module changes the main menu and the way to click it programmatically. This file allowed post_install tests to work.

I don't think so. in module `web_pivot_computed_measure`, it has  [click o_navbar_apps_menu](https://github.com/OCA/web/blob/a53a35687fc05c3cabd88cbdf1507b15946ff6fb/web_pivot_computed_measure/static/src/test/test.esm.js#L11) step, but github ci test still passes. I think we don't need it anymore because of this [css](https://github.com/OCA/web/blob/1906d5636cb209ebc828f2540e989014a1b08982/web_responsive/static/src/components/apps_menu/apps_menu.scss#L47)

chaule97 commented:
> There's already #3037 doing the migration.

Hi @pedrobaeza , compare to #3037:
- my PR only fix attributes of [web.NavBar.AppsMenu](https://github.com/OCA/web/blob/1906d5636cb209ebc828f2540e989014a1b08982/web_responsive/static/src/components/apps_menu/apps_menu.xml#L8), `3037` replace this [web.NavBar.AppsMenu](https://github.com/OCA/web/blob/e6b6fcf2f9901db378eca3a1c0dcff867eb3cef6/web_responsive/static/src/components/apps_menu/apps_menu.xml#L8)
- my PR fix clean code, example: [1](https://github.com/OCA/web/commit/1906d5636cb209ebc828f2540e989014a1b08982#diff-9aa5f9e92a967de6a791e7a800c3f6a2d2a4cb0de8674b89e615c876d8e1a5b4R18), [2](https://github.com/OCA/web/commit/1906d5636cb209ebc828f2540e989014a1b08982#diff-df26859313a976b44590cfcbe2352470f132f86072da9f5a6bea4f08fc7efae6L78), [3](https://github.com/OCA/web/commit/1906d5636cb209ebc828f2540e989014a1b08982#diff-df26859313a976b44590cfcbe2352470f132f86072da9f5a6bea4f08fc7efae6L112), [4](https://github.com/OCA/web/commit/1906d5636cb209ebc828f2540e989014a1b08982#diff-df26859313a976b44590cfcbe2352470f132f86072da9f5a6bea4f08fc7efae6L135)
- `Control panel buttons use icons to save space.` function is removed 
![image](https://raw.githubusercontent.com/OCA/web/17.0/web_responsive/static/img/form_buttons.gif)

because odoo 18 changed template
![image](https://github.com/user-attachments/assets/c144687b-e3b3-4ce5-b765-a96c339956a2)

- I deleted `web_responsive/static/tests/test_patch.js` because it is not necessary anymore

pilarvargas-tecnativa commented:
Hello! I think the correct way to proceed is to make the appropriate corrections in the existing PR and if they are not attended by the author then open a new PR from the existing one. For the time being, the older PR is still being taken care of and has priority over this one. Please, can you provide your revisions?

chaule97 commented:
> Hello! I think the correct way to proceed is to make the appropriate corrections in the existing PR and if they are not attended by the author then open a new PR from the existing one. For the time being, the older PR is still being taken care of and has priority over this one. Please, can you provide your revisions?

It's my task, but I still encourage him to research my updates.

kobros-tech commented:
@chaule97 
@pedrobaeza 
@pilarvargas-tecnativa 

I like that @chaule97 isolated this module from the error appearing because of other module.
I will compare both, if this one is already includes necessary updates without any issue, I don't mind to merge.

Give me a few to review and come back...

pedrobaeza reviewed (changes requested):
This migration is losing 2 features that are required for me:

- The non mandatory preview of the main attachment and the preview bar sided with possibility of maximizing the preview.
- The previous menu search, because the old one was more flexible typing any part of the menu path. Look here:

  ![imagen](https://github.com/user-attachments/assets/ed5223e7-726f-4ed5-8bf0-caf21960b656)

  If I type an extra `p`, no menu entry is found, where it should:

  ![imagen](https://github.com/user-attachments/assets/d8ddc8d7-e2e3-4a44-a271-b9d8e4ca5094)

kobros-tech commented:
> > web_responsive/static/tests/test_patch.js
> 
> This file was necessary for testing engine in general. Odoo testing framework in part of browser integration tests (post_install) make clicks. And it also wants to click on the main menu. But web_responsive module changes the main menu and the way to click it programmatically. This file allowed post_install tests to work.

No need to remove the file, I agree with you!
Instead of fixing the cause of issue he escaped by removing the test case, could have better at least delete the model and its related fields as they are only for test cases and no longer needed in the test database.

In fact I preferred a better solution to test already existing models like res.partner
There is a PR of the fix of the test case #3038 I asked @pedrobaeza to talk with people and get it merged or even delete the demo model after fishing the test there, I don't prefer this solution!

Now this module does not need any change in its test cases, I did not change!
For @chaule97 he made entire change, he could have pulled my branch and add only necessary changes 

Now I will review the changes, and welcome to have previews there but once I finish with his commit.

Only what I need now is this #3038 to be merged.

kobros-tech commented:
among all my review, you have good js to format but we are a community!

I write, then you, then someone else. Writing on top of previous is valuable in issue solving.

I hade to inherit with replacement as the thing was taking size even if I hide it. I whish if I can take some you your changes but you didn't help me to choose the most important changes.

There is a syntax to assign classes the modern way, I like it but it still not solving an issue except for nice looking.

chaule97 commented:
> among all my review, you have good js to format but we are a community!
> 
> I write, then you, then someone else. Writing on top of previous is valuable in issue solving.
> 
> I hade to inherit with replacement as the thing was taking size even if I hide it. I whish if I can take some you your changes but you didn't help me to choose the most important changes.
> 
> There is a syntax to assign classes the modern way, I like it but it still not solving an issue except for nice looking.


Sorry, I am so busy. I will review your PR tomorrow

pedrobaeza commented:
According to comments, closing this one for adopting finally #3037
